BE/Domain/KMS.Product.Ktm.Services/SlackService/SlackFixture.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Microsoft.Extensions.Configuration;
+using Polly;
+using SlackAPI;
+using SlackAPI.WebSocketMessages;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+
+namespace KMS.Product.Ktm.Services.SlackService
+{
+    internal class SlackFixture : IDisposable
+    {
+        private const int MaxConnectionAttempts = 5;
+        private readonly Lazy<SlackSocketClient> botClient;
+        private readonly Lazy<SlackTaskClient> botClientAsync;
+        private readonly Policy connectRetryPolicy;
+        private readonly Lazy<SlackSocketClient> userClient;
+        private readonly Lazy<SlackTaskClient> userClientAsync;
+        public Dictionary<string, User> Users { get; set; }
+        public SlackConfig Config { get; }
+        private SlackFixture(IConfiguration configuration)
+        {
+            this.Config = this.GetHostConfig(configuration);
+
+            this.connectRetryPolicy = Policy
+                .Handle<InvalidOperationException>(exception => exception.Message.Contains("ratelimited"))
+                .WaitAndRetry(MaxConnectionAttempts, retryAttempt => TimeSpan.FromSeconds(ComputeExponentialBackoff(retryAttempt)),
+                    (exception, timeSpan, retryCount, context) => Console.WriteLine($"Connection failed ({exception.Message}). Retrying after {timeSpan.TotalSeconds}s ({retryCount}/5)"));
+
+            this.userClient = new Lazy<SlackSocketClient>(() => connectRetryPolicy.Execute(() => this.CreateClient(this.Config.UserAuthToken)));
+            this.botClient = new Lazy<SlackSocketClient>(() => connectRetryPolicy.Execute(() => this.CreateClient(this.Config.BotAuthToken)));
+            this.userClientAsync = new Lazy<SlackTaskClient>(() => new SlackTaskClient(this.Config.UserAuthToken));
+            this.botClientAsync = new Lazy<SlackTaskClient>(() => new SlackTaskClient(this.Config.BotAuthToken));
+            FetchData();
+        }
+        private void FetchData()
```

Comment:
Please add blank line between each functions/code block.
Please review and apply for other places too.